src/coreclr/gc/gc.cpp

```diff
@@ -18285,7 +18298,38 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
             n = max_generation;
             *blocking_collection_p = TRUE;
             settings.loh_compaction = TRUE;
-            dprintf (GTC_LOG, ("compacting LOH due to hard limit"));
+            dprintf (GTC_LOG, ("compacting gen2+loh due to hard limit"));
+        }
+    }
+
+    if ((conserve_mem_setting != 0) && (n >= max_generation))
```

Comment:
n cannot be > max_generation... it can only be at most max_generation.

Reply:
I know - I changed the >= to ==.